markdown/how.md

```diff
@@ -3,17 +3,68 @@
 
 ## OpenStack
 
+<!-- Note -->
+
 
 ## Magnum
 
+<!-- Note -->
+
+In order to transition to the Docker-based deployment model with Kubernetes, we required a container orchestration service and a container registry offering for storing private container images.
+Within our cloud infrastructure, we had two options available for container orchestration services: Rancher and Magnum. We chose OpenStack Magnum because it provided all the necessary components for a production-ready setup for deploying Tutor.
+
 
 ### Private registry
 
+<!-- Note -->
+
+We needed a container orchestration service that could accommodate custom-built Docker images, and thus we decided to utilize the private Docker registry provided by Magnum. The built-in Docker registry runs locally on each Kubernetes's worker nodes localhost. It does not require authentication, making it easy to publish and pull images without any configuration changes to Tutor.
+
+The Docker registry supported by Magnum utilizes Swift for storage and does not maintain any local state. Therefore, it allows us to launch it as needed in any CI environment,as long as we can provide the necessary Swift credentials.
+
 
 ## Kubernetes
 
+<!-- Note -->
+
+Octavia is the OpenStack load balancing service. We create a Kubernetes cluster with Magnum and use an Octavia Load Balancer to access it from the public network, via the the load balancer’s floating IP.
+
+Magnum lets you create clusters via the OpenStack APIs. To do that, you base your configuration on a cluster template.
+The template defines paramters how the cluster will be constructed.
+
+We deploy our kubernetes cluster with the following configuration.
+In our production environment, we have six node kubernetes cluster three control plane nodes and three worker nodes.
+
+We deploy our cluster with kubernetes version as v.1.21.10 on Fedora CoreOS 34 with Cinder csi driver enabled so that we can provision OpenStack cinder volumes dynamically. Openstack cinder is a block storage service that allows you to create persistent volumes for your kubernetes cluster so when you enable cinder csi driver on your magnum cluster you can create persistent volumes backed by cinder block storage.
+
 
 ## Ceph
 
+<!-- Note -->
+
+To achieve automatic replication between the primary and DR site, we use Backup and Restore policy using the Ceph object gateway S3 API. 
+
+In our kubernetes environment, the backup and restore process is implemented using cron jobs. The cron job is responsible for capturing MySQL and MongoDB dumps from the primary site, uploading them to S3 buckets, and subsequently restoring them from these S3 buckets at the DR site.  
```

Comment:
Here, you'll want to briefly explain what a CronJob is in a Kubernetes context. Not too many people are aware that this resource type even exists, much less what it can be used for.